### PR TITLE
Make partial to improve theming of data consent

### PR DESCRIPTION
**Pull request by BessieSteinberg**
Move content of the Data Sharing Consent agreement to a partial so that
the form's language can be themed without effecting the rest of the page

**Description:** Describe in a couple of sentence what this PR adds

**JIRA:** Link to JIRA ticket

**Dependencies:** dependencies on other outstanding PRs, issues, etc. 

**Merge deadline:** List merge deadline (if any)

**Installation instructions:** List any non-trivial installation 
instructions.

**Testing instructions:**

1. Open page A
2. Do thing B
3. Expect C to happen
4. If D happend instead - check failed.

**Merge checklist:**

- [ ] Check that the versions of the requirements in the `platform-master.in` file match edx-platform.
- [ ] New requirements are in the right place (`base.in` if only used in enterprise; in the correct `platform-****.in` files if they're hosted in edx-platform)
- [ ] Regenerate requirements with `make upgrade && make requirements` (and make sure to fix any errors).
  **DO NOT** just add dependencies to `requirements/*.txt` files.
- [ ] Called `make static` for webpack bundling if any static content was updated.
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are (reasonably) squashed
- [ ] Translations are updated
- [ ] PR author is listed in AUTHORS

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is finished.
- [ ] Delete working branch (if not needed anymore)
- [ ] edx-platform PR (be sure to include edx-platform requirements upgrades that were present in this PR)

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.

### CODE REVIEW

File: enterprise/templates/enterprise/grant_data_sharing_permissions_agreement_content.html [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+                </header>
+                <p>{{ confirmation_alert_prompt|safe }}</p>
+                <p>{{ confirmation_alert_prompt_warning }}</p>
+                <button class="consent-agreement-button" id="modal-no-consent-button">{{ confirmation_modal_affirm_decline_text }}</button>
+                <button class="failure-link" id="review-policy-link">{{ confirmation_modal_abort_decline_text }}</button>
+              </div>
+            </div>
+          </div>
+        </div>
+      </main>
```
Comment (douglashall) [inline enterprise/templates/enterprise/grant_data_sharing_permissions_agreement_content.html]: Add an extra line to the end of the file to git rid of that red mark.

File: enterprise/templates/enterprise/grant_data_sharing_permissions_agreement_content.html [hunk 2]
```diff
@@ -0,0 +1,109 @@
+{% load i18n staticfiles enterprise %}
+
+<main>
```
Comment (douglashall) [inline enterprise/templates/enterprise/grant_data_sharing_permissions_agreement_content.html]: Fix the indentation below.

File: enterprise/templates/enterprise/grant_data_sharing_permissions.html [hunk 3]
```diff
@@ -27,113 +27,7 @@
...
-                        <p>{{ confirmation_alert_prompt|safe }}</p>
-                        <p>{{ confirmation_alert_prompt_warning }}</p>
-                        <button class="consent-agreement-button" id="modal-no-consent-button">{{ confirmation_modal_affirm_decline_text }}</button>
-                        <button class="failure-link" id="review-policy-link">{{ confirmation_modal_abort_decline_text }}</button>
-                      </div>
-                    </div>
-                  </div>
-                </div>
-              </main>
+              {% include "enterprise/grant_data_sharing_permissions_agreement_content.html" %}
```
Comment (douglashall) [inline enterprise/templates/enterprise/grant_data_sharing_permissions.html]: Can you name the partial `_data_sharing_policy.html`? The leading `_` is a convention to indicate that a template is included in another template.

Comment (BessieSteinberg) [reply enterprise/templates/enterprise/grant_data_sharing_permissions.html]: yes, I'll make that change

**Comment by douglashall** [comment]
@BessieSteinberg Looks like you forgot to bump the version here.

**Pull request merged**